bigchaindb/consensus.py

```diff
@@ -1,244 +1,24 @@
-
-    A consensus plugin must expose a class inheriting from this one via an
-    entry_point.
-
-    All methods listed below must be implemented.
-    """
-
-    @staticmethod
-    @abstractmethod
-    def validate_transaction(bigchain, transaction):
-        """Validate a transaction.
-
-        Args:
-            bigchain (Bigchain): an instantiated ``bigchaindb.Bigchain`` object.
-            transaction (dict): transaction to validate.
-
-        Returns:
-            The transaction if the transaction is valid else it raises an
-            exception describing the reason why the transaction is invalid.
-
-        Raises:
-            Descriptive exceptions indicating the reason the transaction failed.
-            See the `exceptions` module for bigchain-native error classes.
-        """
-
-    @staticmethod
-    @abstractmethod
-    def validate_block(bigchain, block):
-        """Validate a block.
-
-        Args:
-            bigchain (Bigchain): an instantiated ``bigchaindb.Bigchain`` object.
-            block (dict): block to validate.
-
-        Returns:
-            The block if the block is valid else it raises an exception
-            describing the reason why the block is invalid.
-
-        Raises:
-            Descriptive exceptions indicating the reason the block failed.
-            See the `exceptions` module for bigchain-native error classes.
-        """
-
-    @staticmethod
-    @abstractmethod
-    def create_transaction(*args, **kwargs):
-        """Create a new transaction.
-
-        Args:
-            The signature of this method is left to plugin authors to decide.
-
-        Returns:
-            dict: newly constructed transaction.
-        """
-
-    @staticmethod
-    @abstractmethod
-    def sign_transaction(transaction, *args, **kwargs):
-        """Sign a transaction.
-
-        Args:
-            transaction (dict): transaction to sign.
-            any other arguments are left to plugin authors to decide.
-
-        Returns:
-            dict: transaction with any signatures applied.
-        """
-
-    @staticmethod
-    @abstractmethod
-    def validate_fulfillments(signed_transaction):
-        """Validate the fulfillments of a transaction.
-
-        Args:
-            signed_transaction (dict): signed transaction to verify
-
-        Returns:
-            bool: True if the transaction's required fulfillments are present
-                and correct, False otherwise.
-        """
-
-    @abstractmethod
-    def verify_vote_signature(block, signed_vote):
-        """Verify a cast vote.
-
-        Args:
-            block (dict): block under election
-            signed_vote (dict): signed vote to verify
-
-        Returns:
-            bool: True if the votes's required signature data is present
-                and correct, False otherwise.
-        """
-
-
-class BaseConsensusRules(AbstractConsensusRules):
+class BaseConsensusRules():
```

Comment:
I kind of agree with @sohkai in the sense that functions would be sufficient. The module acts as a namespace. In any case I would prefer that we treat that in a separate issue, as the current solution works, and this PR is already big enough, and has been time-consuming.